api/v3rpc/rpctypes/error.go

```diff
@@ -277,3 +279,25 @@ func ErrorDesc(err error) string {
 	}
 	return err.Error()
 }
+
+func ContextError(ctx context.Context, err error) error {
```

Comment:
Need to decide where to put this function, we either expose it from clientv3, or move to api as proposed here.

rpctypes package seems a good place as it already has functions to extract errors from etcd rpc.